client/packages/system/src/ProgramEnrolment/ListView/ListView.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { FC, useMemo } from 'react';
+import {
+  TableProvider,
+  DataTable,
+  useColumns,
+  createTableStore,
+  NothingHere,
+  createQueryParamsStore,
+  useFormatDateTime,
+  ColumnAlign,
+  useUrlQueryParams,
+  useNavigate,
+} from '@openmsupply-client/common';
+import { useProgramEnrolment } from '../api/hooks';
+import { ProgramFragment } from '../api/operations.generated';
+import { AppBarButtons } from './AppBarButtons';
+
+type ProgramFragmentWithId = { id: string } & ProgramFragment;
+
+const ProgramListComponent: FC = () => {
+  const {
+    updateSortQuery,
+    updatePaginationQuery,
+    queryParams: { sortBy, page, first, offset },
+  } = useUrlQueryParams();
+  const { data, isError, isLoading } = useProgramEnrolment.document.list();
+  const dataWithId: ProgramFragmentWithId[] | undefined = useMemo(
+    () =>
+      data?.nodes.map(node => ({
+        id: node.name,
+        ...node,
+      })),
+    [data]
+  );
+  const pagination = { page, first, offset };
+  const { localisedDate } = useFormatDateTime();
+
+  const columns = useColumns<ProgramFragmentWithId>(
+    [
+      {
+        key: 'type',
+        label: 'label.enrolment-program',
+      },
+      {
+        key: 'programPatientId',
+        label: 'label.enrolment-patient-id',
+      },
+      {
+        key: 'enrolmentDatetime',
+        label: 'label.enrolment-datetime',
+        align: ColumnAlign.Right,
+        width: 160,
+        formatter: dateString =>
+          dateString ? localisedDate((dateString as string) || '') : '',
+      },
+    ],
+    { onChangeSortBy: updateSortQuery, sortBy },
+    [sortBy]
+  );
+  const navigate = useNavigate();
+
+  return (
+    <>
+      <AppBarButtons />
+      <DataTable
+        key="program-enrolment-list"
+        pagination={{ ...pagination, total: data?.totalCount }}
+        onChangePage={updatePaginationQuery}
+        columns={columns}
+        data={dataWithId}
+        isLoading={isLoading}
+        isError={isError}
+        onRowClick={row => {
+          navigate(
+            `${row.type}?doc=${row.document.name}&patientId=${row.patientId}&type=${row.type}`
```

Comment:
there is already the patient id in the URL - can we use that? This URL is getting huge!
We also have `row.type` twice, can't that be taken from the URL too, rather than the search params?

`http://localhost:3003/patients/patient1/programs/TestProgram1?doc=patients/patient1/programs/TestProgram1&patientId=patient1&type=TestProgram1`

Reply:
agree I created an issue: https://github.com/openmsupply/open-msupply/issues/400